### PR TITLE
Added NIOS operation to allocate network container

### PR DESCRIPTION
**Pull request by Eric-Jckson**
Add Allocate Network Container NIOS operation. This is used to dynamically create network containers to be used with infoblox terraform.

Link to Infoblox Terraform Issue.
infobloxopen/terraform-provider-infoblox#157

### CODE REVIEW

**Comment by Eric-Jckson** [comment]
Hey @somashekhar @anagha-infoblox Please let me know if any changes need to be made on this PR thanks!

**Comment by Eric-Jckson** [comment]
@ranjishmp Please let me know if any changes need to be made on this PR thanks!

**Comment by Eric-Jckson** [comment]
@ranjishmp @somashekhar @anagha-infoblox

Hey will this PR be in review for the upcoming sprint? thanks!

**Comment by Eric-Jckson** [comment]
Hey @ranjishmp ! any update when this might be moved in the sprint?

**Comment by ranjishmp** [comment]
@skudriavtsev Please review this and provide your comments

File: object_manager_network_container.go [hunk 1]
```diff
@@ -76,6 +76,33 @@ func (objMgr *ObjectManager) UpdateNetworkContainer(
...
+	isIPv6 bool,
+	prefixLen uint,
+	comment string,
+	eas EA) (networkContainer *NetworkContainer, err error) {
+
+	containerInfo := NewNetworkContainerNextAvailableInfo(netview, cidr, prefixLen, isIPv6)
+	container := NewNetworkContainerNextAvailable(containerInfo, isIPv6, comment, eas)
+
+	ref, err := objMgr.connector.CreateObject(container)
+	fmt.Println(ref)
```
Comment (skudriavtsev) [inline object_manager_network_container.go:91]: Please, remove this.

File: objects.go [hunk 2]
```diff
@@ -604,7 +655,7 @@ type HostRecordIpv4Addr struct {
 	IBBase     `json:"-"`
 	Ipv4Addr   string `json:"ipv4addr,omitempty"`
 	Ref        string `json:"_ref,omitempty"`
-	Mac        string `json:"mac"`
+	Mac        string `json:"mac,omitempty"`
```
Comment (skudriavtsev) [inline objects.go:658]: Why this is needed? We had issues with making MAC-address field empty (deleting association), that is why this 'omitempty' option was removed.

File: objects.go [hunk 3]
```diff
@@ -636,7 +687,7 @@ type HostRecordIpv6Addr struct {
 	IBBase     `json:"-"`
 	Ipv6Addr   string `json:"ipv6addr,omitempty"`
 	Ref        string `json:"_ref,omitempty"`
-	Duid       string `json:"duid"`
+	Duid       string `json:"duid,omitempty"`
```
Comment (skudriavtsev) [inline objects.go:690]: The same as for 'mac' field.

**Review by skudriavtsev — CHANGES_REQUESTED** [review]
Looks good in overall. Please address my comments, double-check that the code is aligned with the latest changes in the master branch, that all works as expected by you (not only with UTs). After this, I will perform the final review, testing, will approve the PR and merge ASAP.

File: object_manager_network_container.go (hunk 1 above)
Comment (Eric-Jckson) [reply object_manager_network_container.go]: I have removed this debug print statement.

File: objects.go (hunk 2 above)
Comment (Eric-Jckson) [reply objects.go]: This was a new change that i did not have in my branch. I have removed it in my branch.

File: objects.go (hunk 3 above)
Comment (Eric-Jckson) [reply objects.go]: This was a new change that i did not have in my branch. I have removed it in my branch.

**Comment by Eric-Jckson** [comment]
> Looks good in overall. Please address my comments, double-check that the code is aligned with the latest changes in the master branch, that all works as expected by you (not only with UTs). After this, I will perform the final review, testing, will approve the PR and merge ASAP.

Thanks for taking a look at the PR @skudriavtsev . I have made the above changes that you suggested for review. Please let me know if anything else needs to be modified.

**Comment by Eric-Jckson** [comment]
Hey @skudriavtsev , @ranjishmp does anything else need to be updated for this PR?

**Comment by ranjishmp** [comment]
@Eric-Jckson @skudriavtsev will review this next week.

**Comment by Eric-Jckson** [comment]
Hey @ranjishmp @skudriavtsev , any update on this PR?

**Review by skudriavtsev — APPROVED** [review]
Sorry for the late answer.
LGTM

**Comment by Eric-Jckson** [comment]
Thanks @skudriavtsev ! Do you know when the next tag 2.2.0 will be released?

**Pull request merged**